install-mongodb.sh

```diff
@@ -1,20 +1,43 @@
 #!/bin/bash
 set -e
 
-apt-get install gnupg curl >> ~/install-mongo.log 2>&1
-
-curl -fsSL https://pgp.mongodb.com/server-6.0.asc | \
-   gpg -o /usr/share/keyrings/mongodb-server-6.0.gpg \
-   --dearmor >> ~/install-mongo.log 2>&1
-
-echo "deb [ arch=amd64,arm64 signed-by=/usr/share/keyrings/mongodb-server-6.0.gpg ] https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/6.0 multiverse" | tee /etc/apt/sources.list.d/mongodb-org-6.0.list >> ~/install-mongo.log 2>&1
-
-apt-get update >> ~/install-mongo.log 2>&1
-
-apt-get install -y mongodb-org >> ~/install-mongo.log 2>&1
-
-systemctl start mongod >> ~/install-mongo.log 2>&1
-
-systemctl status mongod >> ~/install-mongo.log 2>&1
-
-systemctl enable mongod >> ~/install-mongo.log 2>&1
+USER_NAME="dasadmin"
+MONGO_VERSION="7.0.5"
+MONGO_PORT="28100"
+
+function user_setup() {
+   adduser --disabled-password --gecos "" "$USER_NAME"
+   usermod -a -G $USER_NAME $USER_NAME
+}
+
+function docker_setup() {
+   apt-get -y update
+   apt-get -y install net-tools ca-certificates curl gnupg
+
+   if [ ! -f /etc/apt/keyrings/docker.gpg ]; then
+      install -m 0755 -d /etc/apt/keyrings
+      curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+      chmod a+r /etc/apt/keyrings/docker.gpg
+      echo \
+         "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+        "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" |
+         tee /etc/apt/sources.list.d/docker.list >/dev/null
+   fi
+
+   apt-get -y update
+   apt-get -y install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+   dockerd > /dev/null 2>&1 < /dev/null &
+   docker run hello-world
+   docker rm $(docker ps -a | grep hello | cut -d" " -f1)
+   docker rmi hello-world
+}
+
+function mongo_setup() {
+   docker image pull mongo:$MONGO_VERSION
+
+   docker run --name mongo --restart always -p ${MONGO_PORT}:27017 -d mongo:$MONGO_VERSION
+}
+
+user_setup
+docker_setup
+mongo_setup
```

Comment:
I didn't see where MongoDB port is being set open in ufw. Are we doing it somewhere else?
Anyway, this port should not be open for every IP, but only for the OpenFaas server.